libunwind/src/UnwindRegistersRestore.S

```diff
@@ -731,8 +712,12 @@ WEAK_ALIAS(__rtld_unw_setcontext, _rtld_unw_setcontext)
   .p2align 2
 DEFINE_LIBUNWIND_FUNCTION(__libunwind_Registers_arm64_jumpto)
 #ifdef __CHERI_PURE_CAPABILITY__
+  ldr    c2,      [c0, #0x1f0]  // Pass the target untrusted stack pointer
+  ldr    c3,      [c0, #0x210]  // Pass the target trusted stack pointer
+  bl     dl_c18n_unwind_trusted_stk
```

Comment:
I would much rather not have all this magic "pass through an arbitrary set of arguments as the return values so those registers are preserved, according to whatever the caller this is designed for happens to need" and instead do one of:

1. Follow the ABI of a normal function call

2. Use a highly-customised calling convention like TLSDESC does

I don't think any of the callers of these kinds of functions is performance-critical enough to warrant 2.

Reply:
The latest commit now does 1.